source/common/tls/utility.cc

```diff
@@ -384,6 +384,24 @@ absl::optional<uint32_t> Utility::getDaysUntilExpiration(const X509* cert,
   return absl::nullopt;
 }
 
+std::vector<std::string> Utility::getCertificateExtensionOids(X509& cert) {
+  std::vector<std::string> extension_oids;
+
+  int count = X509_get_ext_count(&cert);
+  for (int pos = 0; pos < count; pos++) {
+    X509_EXTENSION* extension = X509_get_ext(&cert, pos);
+    RELEASE_ASSERT(extension != nullptr, "");
+
+    char oid[256];
```

Comment:
Can this `256` come from any constant from `openssl/ssl.h` or other boringssl header?

Reply:
Seems I couldn't find any RFC standard, or definition from ssl library header files.

I refactored the size definition to a const value, together with a pointer to OpenSSL library. Per https://github.com/openssl/openssl/blob/master/CHANGES.md#changes-between-310-and-311-30-may-2023 `OBJ_obj2txt()` has been capped to `586` bytes.

There are other reference to the value `Utility::generalNameAsString()` from the code base, so here we are keeping the same, smaller safer limit in this PR.